### PR TITLE
MXNet Java bug fixes and experience improvement

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Add visualization method to Java users and source code fetching in demo section
@zachgk @piyushghai @andrewfayres @yzhliu 
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: scala-package/mxnet-demo/java-demo/pom.xml [hunk 1]
```diff
@@ -82,6 +82,12 @@
             <artifactId>mxnet-full_${mxnet.scalaprofile}-${mxnet.profile}</artifactId>
             <version>${mxnet.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.mxnet</groupId>
+            <artifactId>mxnet-full_${mxnet.scalaprofile}-${mxnet.profile}</artifactId>
+            <version>${mxnet.version}</version>
+            <classifier>sources</classifier>
```
Comment (piyushghai) [inline scala-package/mxnet-demo/java-demo/pom.xml]: What's this for ?

Comment (lanking520) [reply scala-package/mxnet-demo/java-demo/pom.xml]: Attaching the sources jar

File: scala-package/mxnet-demo/java-demo/README.md [hunk 2]
```diff
@@ -42,14 +42,14 @@ The `SCALA_PKG_PROFILE` should be chosen from `osx-x86_64-cpu`, `linux-x86_64-cp
 
 
 ## Run
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]: Please fix the first line of the sample's description.
`This is an project created to use Maven-published Scala/Java package with two Java examples.` --> 
```
This is a project demonstrating how to use the Maven published Scala/Java MXNet package. 
The examples provided include:
* NDArray creation
* Object detection using the Inference API
```
Removing the specific number makes it easier to add more examples later, plus listing them tells the reader what to expect without having to read through the whole doc.

Comment (aaronmarkham) [reply scala-package/mxnet-demo/java-demo/README.md]: Also, in the setup section, please define what folder the user is supposed to run the `mvn package` command. I assume it is from this same `java-demo` folder, but please specify.

File: scala-package/mxnet-demo/java-demo/README.md [hunk 3]
```diff
@@ -42,14 +42,14 @@ The `SCALA_PKG_PROFILE` should be chosen from `osx-x86_64-cpu`, `linux-x86_64-cp
...
 The Scala file is being executed using Java. You can execute the helloWorld example as follows:
 ```Bash
 bash bin/java_sample.sh
 ```
 You can also run the following command manually:
 ```Bash
-java -cp $CLASSPATH sample.HelloWorld
+java -cp $CLASSPATH sample.NDArrayCreation
 ```
 However, you have to define the Classpath before you run the demo code. More information can be found in the `java_sample.sh`.
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]:
```suggestion
However, you have to define the Classpath before you run the demo code. More information can be found in `bin/java_sample.sh`.
```

File: scala-package/mxnet-demo/java-demo/README.md [hunk 4]
```diff
@@ -42,14 +42,14 @@ The `SCALA_PKG_PROFILE` should be chosen from `osx-x86_64-cpu`, `linux-x86_64-cp
 
 
 ## Run
-### Hello World
+### NDArrayCreation
 The Scala file is being executed using Java. You can execute the helloWorld example as follows:
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]:
```suggestion
The Scala file is being executed using Java. You can execute the `HelloWorld` example as follows:
```

File: scala-package/mxnet-demo/java-demo/bin/java_sample.sh [hunk 5]
```diff
@@ -17,4 +17,4 @@
 #!/bin/bash
 CURR_DIR=$(cd $(dirname $0)/../; pwd)
 CLASSPATH=$CLASSPATH:$CURR_DIR/target/*:$CLASSPATH:$CURR_DIR/target/dependency/*
-java -Xmx8G  -cp $CLASSPATH mxnet.HelloWorld
\ No newline at end of file
+java -Xmx8G  -cp $CLASSPATH mxnet.NDArrayCreation
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/bin/java_sample.sh]:
```suggestion
java -Xmx8G  -cp $CLASSPATH mxnet.NDArrayCreation

```

**Comment by lanking520** [comment]
@aaronmarkham Please review it again

**Pull request merged**